File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.py
class Solution:
    def maxScore(self, s: str) -> int:
        max_value = 0
        zeros = 0
        ones = 0
        arr = list(s)

        for i in range(1, len(arr)):
            if arr[i] == '1':
                ones += 1

        if arr[0] == '0':
            zeros += 1

        for i in range(1, len(arr)):
            sum_value = zeros + ones
            max_value = max(sum_value, max_value)
            if arr[i] == '0':
                zeros += 1
            else:
                ones -= 1

        return max_value

File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
var maxScore = function (s) {
    let max = 0
    let zeros = 0
    let ones = 0
    let arr = s.split("")
    for (let i = 1; i < arr.length; i++) {
        if (arr[i] == 1) {
            ones += 1
        }
    }
    if (arr[0] == 0) {
        zeros += 1
    }
    for (let i = 1; i < arr.length ; i++) {
        let sum = zeros + ones
        max = Math.max(sum, max)
        if (arr[i] == 0) {
            zeros += 1
        }
        else {
            ones -= 1
        }
    }
    return max
};